doc/cabal-project.rst

```diff
@@ -1333,156 +1336,140 @@ Haddock options
-    ``haddock`` subcommand).
-
 .. cfg-field:: haddock-tests: boolean
+               --haddock-tests
     :synopsis: Generate documentation for tests.
 
     :default: False
 
     Run haddock on all test suites.
 
-    The command line variant of this flag is ``--tests`` (for the
-    ``haddock`` subcommand).
-
 .. cfg-field:: haddock-benchmarks: boolean
+               --haddock-benchmarks
     :synopsis: Generate documentation for benchmarks.
 
     :default: False
 
     Run haddock on all benchmarks.
 
-    The command line variant of this flag is ``--benchmarks`` (for the
-    ``haddock`` subcommand).
-
-.. cfg-field:: haddock-all: boolean
-    :synopsis: Generate documentation for everything
-
-    :default: False
-
-    Run haddock on all components.
-
-    The command line variant of this flag is ``--all`` (for the
-    ``haddock`` subcommand).
-
 .. cfg-field:: haddock-internal: boolean
+               --haddock-internal
     :synopsis: Generate documentation for internal modules
 
     :default: False
 
     Build haddock documentation which includes unexposed modules and
     symbols.
 
-    The command line variant of this flag is ``--internal`` (for the
-    ``haddock`` subcommand).
+.. cfg-field:: haddock-all: boolean
+               --haddock-all
+    :synopsis: Generate documentation for everything
+
+    :default: False
+
+    Run haddock on all components.
 
 .. cfg-field:: haddock-css: path
-    :synopsis: Location of Haddoc CSS file.
+               --haddock-css=PATH
+    :synopsis: Location of Haddock CSS file.
 
     The CSS file that should be used to style the generated
-    documentation (overriding haddock's default.)
-
-    The command line variant of this flag is ``--css`` (for the
-    ``haddock`` subcommand).
+    documentation (overriding haddock's default).
 
 .. cfg-field:: haddock-hyperlink-source: boolean
+               --haddock-hyperlink-source
     :synopsis: Generate hyperlinked source code for documentation
 
     :default: False
 
     Generated hyperlinked source code using `HsColour`_, and have
     Haddock documentation link to it.
-
-    The command line variant of this flag is ``--hyperlink-source`` (for
-    the ``haddock`` subcommand).
+    This is equivalent to running ``haddock`` with the ``--hyperlinked-source`` flag.
 
 .. cfg-field:: haddock-hscolour-css: path
+               --haddock-hscolour-css=PATH
     :synopsis: Location of CSS file for HsColour
 
     The CSS file that should be used to style the generated hyperlinked
     source code (from `HsColour`_).
 
-    The command line variant of this flag is ``--hscolour-css`` (for the
-    ``haddock`` subcommand).
-
 .. cfg-field:: haddock-contents-location: URL
+               --haddock-contents-location=URL
     :synopsis: URL for contents page.
 
     A baked-in URL to be used as the location for the contents page.
 
-    The command line variant of this flag is ``--contents-location``
-    (for the ``haddock`` subcommand).
-
```

Comment:
Maybe add the same explicit note (about the absence of a command-line equivalent) that you have for keeping temporary files below?

Reply:
It is not absent:
```
 --haddock-contents-location=URL
                                Bake URL in as the location for the contents
                                page
```